src/components/splitButton/splitButton.ts

```diff
@@ -0,0 +1,410 @@
+        keyCode == UP_ARROW_KEY_CODE;
+
+      if (openDropdown) {
+        this.open = true;
+        this.options[highlightedIndex].highlighted = true;
+
+        // scroll to highlighted option
+        // if (!this.multiple && this.value !== '') {
+        //   this.options[highlightedIndex].scrollIntoView({ block: 'nearest' });
+        // }
+      }
+    }
+    switch (keyCode) {
+      case ENTER_KEY_CODE: {
+        this.assistiveText = 'Selected an item.';
+        return;
+      }
+      case DOWN_ARROW_KEY_CODE: {
+        // go to next option
+        let nextIndex =
+          !highlightedEl && !selectedEl
+            ? 0
+            : highlightedIndex === this.options.length - 1
+            ? 0
+            : highlightedIndex + 1;
+
+        // skip disabled options
+        if (this.options[nextIndex].disabled) {
+          nextIndex = nextIndex === this.options.length - 1 ? 0 : nextIndex + 1;
+        }
+
+        this.options[highlightedIndex].highlighted = false;
+        this.options[nextIndex].highlighted = true;
+
+        // scroll to option
+        this.options[nextIndex].scrollIntoView({ block: 'nearest' });
+
+        this.assistiveText = this.options[nextIndex].text;
+        return;
+      }
+      case UP_ARROW_KEY_CODE: {
+        // go to previous option
+        let nextIndex =
+          highlightedIndex === 0
+            ? this.options.length - 1
+            : highlightedIndex - 1;
+
+        // skip disabled options
+        if (this.options[nextIndex].disabled) {
+          nextIndex = nextIndex === 0 ? this.options.length - 1 : nextIndex - 1;
+        }
+
+        this.options[highlightedIndex].highlighted = false;
+        this.options[nextIndex].highlighted = true;
+
+        // scroll to option
+        this.options[nextIndex].scrollIntoView({ block: 'nearest' });
+
+        this.assistiveText = this.options[nextIndex].text;
+        return;
+      }
+      case ESCAPE_KEY_CODE: {
+        // close listbox
+        this.open = false;
+        this.buttonEl.focus();
+        this.assistiveText = 'Split button menu options.';
+        return;
+      }
+      default: {
+        return;
+      }
+    }
+  }
+
+  private toggleDropdown = () => {
+    this.open = !this.open;
+  };
+
+  // When click outside, then menu close
+  private _handleClickOut(e: Event): void {
+    if (!e.composedPath().includes(this)) {
+      this.open = false;
+    }
+  }
+  // handle child blur event
+  private _handleBlur(e: any): void {
+    const relatedTarget = e.detail.origEvent.relatedTarget;
+
+    if (
+      !relatedTarget ||
+      (relatedTarget.localName !== 'kd-splitbutton-option' &&
+        relatedTarget.localName !== 'kd-split-btn')
+    ) {
+      this.open = false;
+    }
+  }
+
+  private _handleClick(e: any) {
+    this.open = false;
+    // emit selected value
+    this.emitValue(e);
+  }
+
+  private emitValue(e: any) {
+    const event = new CustomEvent('on-change', {
```

Comment:
I don't think `on-change` is really the correct name for this event. Maybe it should also be `on-click`? You would have to call `e.stopPropagation()` on the child element `on-click` in that case since that bubbles. Actually thinking about it more, we should probably always stopPropagation when capturing custom events that bubble.

Reply:
I didn't get this where to put **e.stopPropagation();** ? if we give **on-click** name then should change child element event name something like **on-action-click**

Reply:
can we remove parent on-click as it no longer make sense